iothub/service/src/JobClient/JobClient.cs

```diff
@@ -41,6 +48,62 @@ public static JobClient CreateFromConnectionString(string connectionString, Http
             return new HttpJobClient(iotHubConnectionString, transportSettings);
         }
 
+#if !NET451
+
+        /// <summary>
+        /// Creates an instance of <see cref="JobClient"/>.
+        /// </summary>
+        /// <param name="hostName">IoT hub host name.</param>
+        /// <param name="credential">Azure Active Directory credentials to authenticate with IoT hub. See <see cref="TokenCredential"/></param>
+        /// <param name="transportSettings">The HTTP transport settings.</param>
+        /// <returns>An instance of <see cref="JobClient"/>.</returns>
+        public static JobClient Create(
+            string hostName,
+            TokenCredential credential,
+            HttpTransportSettings transportSettings = default)
+        {
+            if (string.IsNullOrEmpty(hostName))
+            {
+                throw new ArgumentNullException($"{nameof(hostName)},  Parameter cannot be null or empty");
```

Comment:
If you want to include a messag, please use the `ArgumentNullException(string, string)` overload. The first parameter is the null variable name, and the second is the message.

Otherwise the `ArgumentNullException(string)` overload just expects a parameter.

So either of these:
```chsarp
throw new ArgumentNullException(nameof(hostName));
// or
throw new ArgumentNullException(nameof(hostName), $"{Parameter cannot be null or empty");
```

I recommend the former if the message doesn't add anything. For instance, for `hostName` it also can't be empty, so the message adds something. For `credential`, I'd stick with the 1 param option.